theano/tensor/elemwise.py

```diff
@@ -504,7 +504,7 @@ def __setstate__(self, d):
         self.nfunc = None
         self.inplace_pattern = frozendict(self.inplace_pattern)
         if getattr(self, 'nfunc_spec', None):
-            self.nfunc = getattr(numpy, self.nfunc_spec[0])
+            self.nfunc = getattr(np, self.nfunc_spec[0])
         elif 0 < self.scalar_op.nin < 32:
             self.ufunc = np.frompyfunc(self.scalar_op.impl,
                                           self.scalar_op.nin,
```

Comment:
IMO, the best fix for this situation would be, squash into one commit, move to a different branch(branching out from master), then cherry pick this single squashed commit to that branch. I would advise you to squash *only*  the commits that are adding imports. if you are doing something else in this PR, please keep it as a different commit.